### PR TITLE
[#541] making learner readonly for nations (connects #541)

### PR DESCRIPTION
**Pull request by lmmrssa**

### CODE REVIEW

**Comment by lmmrssa** [comment]
added the function in validation rule to not allow user to update. As validation is called only on new entry and update it will invalidate on write @paulbert could you review shall we go this way

File: design/nations/nation-validators.js [hunk 1]
```diff
@@ -5,6 +5,24 @@ function (newDoc, oldDoc, userCtx, secObj) {
...
   }
+  function canWrite() {
+    var CAN_WRITE = true;
+    var readOnly = ['learner'];
+    if (!userCtx.isUserAdmin) {
+      if (userCtx.roles.length < 1) {
+        CAN_WRITE = false;
+      } else {
+        for (var i = 0; i < userCtx.roles.length; i++) {
+          if (~ readOnly.indexOf(userCtx.roles[i])) {
```
Comment (paulbert) [inline design/nations/nation-validators.js]: This is quicker to understand with === -1

File: design/nations/nation-validators.js [hunk 2]
```diff
@@ -5,6 +5,24 @@ function (newDoc, oldDoc, userCtx, secObj) {
...
+            CAN_WRITE = false;
+          }
+        }
+      }
+    }
+    if(!CAN_WRITE)
```
Comment (paulbert) [inline design/nations/nation-validators.js]: Even for a one line block, it's better (more readable, doesn't risk someone writing another line & popping that statement out of the if clause) to have brackets.

File: design/nations/nation-validators.js [hunk 3]
```diff
@@ -5,6 +5,24 @@ function (newDoc, oldDoc, userCtx, secObj) {
       throw { forbidden: errMessage };
     }
   }
+  function canWrite() {
+    var CAN_WRITE = true;
+    var readOnly = ['learner'];
+    if (!userCtx.isUserAdmin) {
```
Comment (paulbert) [inline design/nations/nation-validators.js]: I don't think user profile properties are on a userCtx object.

Comment (lmmrssa) [reply design/nations/nation-validators.js]: had tested it and it was working by that code.

File: design/nations/nation-validators.js (hunk 2 above)
Comment (lmmrssa) [reply design/nations/nation-validators.js]: oops yes i had noticed it but forgot to push that change :) thanks for catching

**Pull request merged**